src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerRequest.Managed.cs

```diff
@@ -596,25 +600,24 @@ public string UserHostName
 
         public string[] UserLanguages
         {
-            get { return _user_languages; }
+            get { return _userLanguages; }
         }
 
         public IAsyncResult BeginGetClientCertificate(AsyncCallback requestCallback, object state)
         {
-            if (_gccDelegate == null)
-                _gccDelegate = new GCCDelegate(GetClientCertificate);
-            return _gccDelegate.BeginInvoke(requestCallback, state);
+            ListenerClientCertAsyncResult result = new ListenerClientCertAsyncResult(GetClientCertificate(), requestCallback, state);
```

Comment:
This is doing the operation synchronously even though the caller wants it to be done asynchronously.  If there's no real way to do it with async I/O (I've not looked to see what the operation is), you can either use Task.Run to invoke GetClientCertificate and store the result/exception into the ListenerClientCertAsyncResult, or you can use Task.Run to invoke GetClientCertificate and then use the TaskToApm internal helper class to get appropriate Begin/End methods for it.